### PR TITLE
Fix sample reading

### LINKED ISSUE #35: Fix low accuracy
We probably have an error in one of our layers, or the optimizers?

### PR DESCRIPTION
**Pull request by Jopyth**
Fixes #35 

I also added some tests (the first sample is compared to a saved png) and normalization.

### CODE REVIEW

**Comment by Jopyth** [comment]
After first epoch with adam:
```
train: epoch: 00, loss: 02.30, accuracy 0.12, iteration: 000
train: epoch: 00, loss: 00.38, accuracy 0.89, iteration: 050
train: epoch: 00, loss: 00.22, accuracy 0.91, iteration: 100
train: epoch: 00, loss: 00.27, accuracy 0.92, iteration: 150
train: epoch: 00, loss: 00.17, accuracy 0.95, iteration: 200
train: epoch: 00, loss: 00.50, accuracy 0.84, iteration: 250
train: epoch: 00, loss: 00.22, accuracy 0.97, iteration: 300
train: epoch: 00, loss: 00.12, accuracy 0.95, iteration: 350
train: epoch: 00, loss: 00.11, accuracy 0.95, iteration: 400
train: epoch: 00, loss: 00.12, accuracy 0.94, iteration: 450
train: epoch: 00, loss: 00.34, accuracy 0.89, iteration: 500
train: epoch: 00, loss: 00.20, accuracy 0.97, iteration: 550
train: epoch: 00, loss: 00.12, accuracy 0.95, iteration: 600
train: epoch: 00, loss: 00.11, accuracy 0.97, iteration: 650
train: epoch: 00, loss: 00.22, accuracy 0.94, iteration: 700
train: epoch: 00, loss: 00.14, accuracy 0.95, iteration: 750
train: epoch: 00, loss: 00.19, accuracy 0.94, iteration: 800
train: epoch: 00, loss: 00.05, accuracy 0.98, iteration: 850
train: epoch: 00, loss: 00.17, accuracy 0.92, iteration: 900
running test set...
 test: epoch: 00, loss: 00.14, accuracy 0.97
```

File: length/tests/data_set_tests/test_data_sets.py [hunk 1]
```diff
@@ -1,12 +1,42 @@
+import numpy as np
+import os
+from PIL import Image
+
+from length.constants import DTYPE
 from length.data_sets import Mnist
 
 
+def test_mnist_with_scale():
+    image = Image.open(os.path.join("length", "tests", "data", "5.png"))
```
Comment (Bartzi) [inline length/tests/data_set_tests/test_data_sets.py]: please use a `with` clause

File: length/tests/data_set_tests/test_data_sets.py [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
+    data_set = Mnist(1, sample_dimensions=2, scale=scale)
+    for batch in data_set.train:
+        first_sample, first_label = batch.data.data[0], batch.labels.data[0]
+        np.testing.assert_almost_equal(expected, first_sample)
+        assert first_label == 5
+        break
+
+
+def test_mnist_without_scale():
+    image = Image.open(os.path.join("length", "tests", "data", "5.png"))
```
Comment (Bartzi) [inline length/tests/data_set_tests/test_data_sets.py]: same here

**Review by Bartzi — APPROVED** [review]
:san_marino:

**Pull request merged**